frontend/express/public/javascripts/countly/countly.vue.components.js

```diff
@@ -372,4 +373,2884 @@
+                    rowKeys = Object.keys(state.patches);
+                }
+                else {
+                    rowKeys = [keyFn(row)];
+                }
+
+                rowKeys.forEach(function(rowKey) {
+                    if (!state.patches[rowKey]) {
+                        return;
+                    }
+
+                    if (!fields) {
+                        Vue.delete(state.patches, rowKey);
+                    }
+                    else {
+                        fields.forEach(function(fieldName) {
+                            Vue.delete(state.patches[rowKey], fieldName);
+                        });
+                        if (Object.keys(state.patches[rowKey]).length === 0) {
+                            Vue.delete(state.patches, rowKey);
+                        }
+                    }
+                });
+
+            }
+        };
+        return VuexModule(name, {
+            resetFn: resetFn,
+            getters: tableGetters,
+            mutations: mutations
+        });
+    };
+
+    var _vuex = {
+        getGlobalStore: function() {
+            return _globalVuexStore;
+        },
+        registerGlobally: function(wrapper, force) {
+            var store = _globalVuexStore;
+            if (!store.hasModule(wrapper.name) || force) {
+                store.registerModule(wrapper.name, wrapper.module);
+            }
+        },
+        Module: VuexModule,
+        DataTable: VuexDataTable
+    };
+
+    var BackboneRouteAdapter = function() {};
+
+    Vue.prototype.$route = new BackboneRouteAdapter();
+
+    var countlyVueWrapperView = countlyView.extend({
+        constructor: function(opts) {
+            this.component = opts.component;
+            this.defaultArgs = opts.defaultArgs;
+            this.vuex = opts.vuex;
+            this.templates = opts.templates;
+            this.elementsToBeRendered = [];
+        },
+        beforeRender: function() {
+            var self = this;
+            if (this.templates) {
+                var templatesDeferred = [];
+                this.templates.forEach(function(item) {
+                    if (typeof item === "string") {
+                        templatesDeferred.push(function(fName) {
+                            return T.get(fName, function(src) {
+                                self.elementsToBeRendered.push(src);
+                            });
+                        }(item));
+                        return;
+                    }
+                    for (var name in item.mapping) {
+                        var fileName = item.mapping[name];
+                        var elementId = item.namespace + "-" + name;
+                        templatesDeferred.push(function(fName, elId) {
+                            return T.get(fName, function(src) {
+                                self.elementsToBeRendered.push("<script type='text/x-template' id='" + elId + "'>" + src + "</script>");
+                            });
```

Comment:
Codacy found an issue: [You should wrap an immediate function invocation in parenthesis.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=6339769)